### PR TITLE
Input Rework

### PR DESCRIPTION
**Pull request by hiddenasbestos**
This commit reworks the input code so that either only virtual joystick-mapped-to-keyboard input is used, or only the actual system keyboard is used. Preventing conflict caused by mapping of keyboard keys to controller inputs.

Primarily changes "User 1 Device Type" to either Virtual Keyboard (default in RA), Keyboard or None.

- **Keyboard** - Read directly from the keyboard only. Pressing TAB activates the on-screen keyboard with cursor key navigation and ENTER to press keys. Right shift has also been added as a mapping for Shift on the system.

- **Virtual Keyboard** - Press SELECT on the pad to toggle the on-screen keyboard, use d-pad to navigate + A or B to press keys. When the on-screen keyboard is hidden, joystick presses may generate key inputs depending on the settings in Core Options.

### CODE REVIEW

**Comment by leiradel** [comment]
I'm sorry, could you elaborate what the problem is? Also, can you resubmit the PR without code format changes? It's hard to follow the changes.

**Comment by leiradel** [comment]
I mean, it seems all spaces at the end of lines were stripped, so the files have changes everywhere instead of only where you actually changed the code.

**Comment by hiddenasbestos** [comment]
Hi, this patch is to simplify the input system from the end user's perspective and resolve the conflict that arises when the keyboard is mapped to joypad controls.

Rather than using two ports it condenses it down to one. It also resolves the issue where the keyboard is mapped to controller inputs which resulted in conflicts when trying to use the keyboard as a real keyboard. This change lets you easily select what input mechanism you'd like to use. It also adds a nice feature where the on-screen keyboard can be operated using the keyboard.

**Comment by leiradel** [comment]
I don't understand. What this changes that could not be achieved before?

**Comment by hiddenasbestos** [comment]
I'm trying to reverse the commit and restage just the changed lines, but git isn't letting me.

**Comment by hiddenasbestos** [comment]
> I don't understand. What this changes that could not be achieved before?
This cleans up the way to select between joystick and keyboard input and removed the bad state of both being enabled.

Keyboard navigation of the on-screen display is new.

**Comment by leiradel** [comment]
> This cleans up the way to select between joystick and keyboard input and removed the bad state of both being enabled.

But we need to be able to enable the keyboard and the joystick. There are joysticks other than Cursor that will be added in the future, such as Kempston.

Or am I missing something?
